python/hsfs/core/training_dataset_engine.py

```diff
@@ -152,6 +171,9 @@ def get_serving_vector(self, training_dataset, entry, external):
         # get schemas for complex features once
         complex_features = self.get_complex_feature_schemas(training_dataset)
 
+        # get transformation functions
+        transformation_fns = self._get_transformation_fns(training_dataset)
```

Comment:
This should happen in the init statement, not here. otherwise the for loop to reload the code is run at every request.